src/main/java/com/dreammaster/gthandler/GT_MachineRecipeLoader.java

```diff
@@ -26575,5 +26577,89 @@ private static final void registerOpticalComponentRecipes() {
                 new ItemStack[] {ItemList.Optical_Cpu_Containment_Housing.get(1L)},
                 10 * 10,
                 7_864_320);
+        
+        // Water Line
+        GT_Values.RA.addSifterRecipe(
+                new ItemStack[] {NI},
+                new FluidStack[] {Materials.Water.getFluid(1000L)},
+                new ItemStack[] {NI},
+                new FluidStack[] {Materials.Grade1PurifiedWater.getFluid(1000L)},
+                500 * 10,
+                32720,
+                true);
+        GT_Recipe.GT_Recipe_Map.sFluidHeaterRecipes.addRecipe(
+                true,
+                new ItemStack[] {NI},
+                null,
+                null,
+                new FluidStack[] {Materials.Grade1PurifiedWater.getFluid(1000L)},
+                new FluidStack[] {Materials.Grade2PurifiedWater.getFluid(1000L)},
+                5000,
+                65440,
+                0);
+        GT_Values.RA.addDistillationTowerRecipe(
+                Materials.Grade2PurifiedWater.getFluid(1000L),
+                new FluidStack[] {
+                    Materials.Grade3PurifiedWater.getFluid(1000L),
+                },
+                null,
+                500 * 10,
+                138880);
+        GT_Values.RA.addLaserEngraverRecipe(
+                new ItemStack[] {GT_ModHandler.getModItem("bartworks", "gt.bwMetaGeneratedlens", 0L, 25)
```

Comment:
I have said this to other. No magic numbers. try to find the ore dict tag and use GT_OredictUnificator.get("oredict", 1)

Reply:
![image](https://user-images.githubusercontent.com/76872108/215319720-475b4b9e-f257-4393-8f08-831cf6a9cc37.png)
You could actually do this for all recipes and use Werkstoff.<MaterialName>.get(OrePrefix.lens, 0). all you have to do is add the if(isModLoaded) and then put inside it the recipes.

Or another alternative is to use BartWorksMaterials#getBartWorksMaterialByIGNName("name"). For this you will need to use GT_OreDictUnificator#get(OrePrefix, Material, Amount)

Reply:
Still required to do the isModLoaded!